neo4j-store/src/main/kotlin/org/ostelco/prime/storage/graph/Neo4jStore.kt

```diff
@@ -404,19 +438,55 @@ object Neo4jStoreSingleton : GraphStore {
                 .ifFailedThenRollback(transaction)
     }
 
-    // TODO vihang: When we read and then delete, it fails when deserialization does not work.
     override fun removeCustomer(identity: org.ostelco.prime.model.Identity): Either<StoreError, Unit> = writeTransaction {
-        write(query = """
-            MATCH (i:${identityEntity.name} {id:'${identity.id}'})-[:${identifiesRelation.name}]->(c:${customerEntity.name})
-            OPTIONAL MATCH (c)-[:${customerToBundleRelation.name}]->(b:${bundleEntity.name})
-            OPTIONAL MATCH (c)-[:${scanInformationRelation.name}]->(s:${scanInformationEntity.name})
-            DETACH DELETE i, c, b, s;
-        """.trimIndent(), transaction = transaction) { statementResult ->
-            Either.cond(
-                    test = statementResult.summary().counters().nodesDeleted() > 0,
-                    ifTrue = {},
-                    ifFalse = { NotFoundError(type = identityEntity.name, id = identity.id) })
-        }
+        IO {
+            Either.monad<StoreError>().binding {
+                // get customer id
+                val customerId = getCustomerId(identity).bind()
+                // create ex-customer with same id
+                create { ExCustomer(id = customerId, terminationDate = LocalDate.now().toString()) }.bind()
+                // get all subscriptions and link them to ex-customer
+                val subscriptions = get(Subscription subscribedBy (Customer withId customerId)).bind()
```

Comment:
`Customer withId customerId` is used in more than one location, move this to a variable